Add test case with creating Message from parameters

mnurzyns opened the pull request:
Due to testing time getting in TimeManagmentTest it's used regex to
ignore problem that we need faketime while creating new message. Now we
test if every kind of message (read, inviter, recipient) is correctly
created and equal to expected value.

Commit:
Add test case with creating Message from parameters

Due to testing time getting in TimeManagmentTest it's used regex to
ignore problem that we need faketime while creating new message. Now we
test if every kind of message (read, inviter, recipient) is correctly
created and equal to expected value.

Commit:
Merge branch 'master' of https://github.com/NMarcin/MessengerOnFileForLinux into mnurzyns

Commit:
Add not comitted change for MessageTest

marcinwozniak0 commented on MessengerOnFileForLinux/MessageFramework/test/MessageTest.cpp:
```diff
@@ -1,11 +1,36 @@
+TEST_F(MessageFixture, isMessageCorrectCreatedFromParameters)
 {
-    EXPECT_TRUE(true);
+         const std::regex regexReadMessage("\\[0]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test read message]");
```
formatowanie

mnurzyns replied:
specjalnie zrobione, bo jest czytelniejsze -> możesz w łatwy sposób porównać sobie 3 regexy, że są prawie takie same.

marcinwozniak0 replied:
chodzi mi o taby przed constem

mnurzyns replied:
mam ci napisać jeszcze raz to samo? Te consty są wyrównane do średników, a nie do lewej linii - powód masz powyżej.

marcinwozniak0 commented on MessengerOnFileForLinux/MessageFramework/test/MessageTest.cpp:
```diff
@@ -1,11 +1,36 @@
+    const std::string strInviterMessage = inviterMessage.messageToSave();
+    const std::string strRecipientMessage = recipientMessage.messageToSave();
+
+    std::cout << strReadMessage << std::endl;
```
do wyrzucenia couty

mnurzyns replied:
ok

marcinwozniak0 commented on MessengerOnFileForLinux/MessageFramework/test/MessageTest.cpp:
```diff
@@ -1,11 +1,36 @@
-    EXPECT_TRUE(true);
+         const std::regex regexReadMessage("\\[0]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test read message]");
+      const std::regex regexInviterMessage("\\[1]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test inviter message]");
+    const std::regex regexRecipientMessage("\\[2]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test recipient message]");
```
nie widzę sensu testowania 3 przypadków. Jeden wystaczy. Co innego jakby sam Message w jakiś sposób ustawiał flagę, ale jest ona podawana w argumencie. Więc skoro działa dla argumentu "0" to tak samo zadziala dla arguemntu "1"

mnurzyns replied:
Nope, nope, nope. 
>     setMessageFlag(messageFlag);
To w c-torze z parametrami, a dalej masz:

> bool Message::setMessageFlag(std::string messageFlag)
> {
>     log_.function("Message::setMessageFlag()");
>     messageFlag_ = messageFlag;
>     if(MessageFlag::inviterMessage == messageFlag || MessageFlag::recipientMessage == messageFlag)
>     {
>         std::string logData = "Message::setMessageFlag() messageFlag_ set as " + messageFlag;
>         log_.info(logData);
>         return true;
>     }
>     else if (MessageFlag::readMessage == messageFlag)
>     {
>         log_.info("Message::setMessageFlag() message read, can not unread");
>         return false;
>     }
>     else
>     {
>         log_.info("Message::setMessageFlag() messageFlag unknown, error");
>         return false;
>     }
> }

marcinwozniak0 replied:
Ok. To zrób z tego test parametryczny,

mnurzyns replied:
Ok, zaakceptuj mi tego pull requesta. A w następnym zmienie wszystkie testy by były parametryczne i tylko wywoływały jakąś inną funkcję.

marcinwozniak0 commented on MessengerOnFileForLinux/scripts/local_scripts/run_messenger.sh:
```diff
@@ -2,7 +2,7 @@
 rm Logger_default.txt
 
```
pamietaj o INT: w nazwie takiego commita.

mnurzyns replied:
1. To było sumowane
2. INT i MSG_1 idzie dopiero od FB_4 bo jeszcze nie mamy choćby tasków ponumerowanych.

Commit:
Add testcases for PurgeMessage test suite

Most important info is that we need ASSERT_EQ in SetUp() due to
we can not test anything if Message is not create correctly.

Commit:
Add test for else return false in Message

Test for checking behaviour of setting incorrect content
and incorrect messageFlag. New task created for hanling it us an
errror, currently is ingored and created as a normal message.
